src/olympia/reviews/views.py

```diff
@@ -419,6 +420,13 @@ def reply(self, *args, **kwargs):
         # FK to the current review object and only allow add-on authors/admins.
         # Call get_object() to trigger 404 if it does not exist.
         self.review_object = self.get_object()
+        if Review.unfiltered.filter(reply_to=self.review_object).exists():
+            # A reply already exists, just edit it.
+            # We set should_access_deleted_reviews so that it works even if
+            # the reply has been deleted.
+            self.kwargs['pk'] = kwargs['pk'] = self.review_object.reply.pk
+            self.should_access_deleted_reviews = True
```

Comment:
I'm less and less a fan of these kinds of side-effects. Can't we just pass that directly to the serializer, e.g in `get_serializer_context`?